sonar-db/src/main/java/org/sonar/db/property/PropertiesDao.java

```diff
@@ -161,6 +164,23 @@ public PropertyDto selectProjectProperty(long resourceId, String propertyKey) {
     return session.getMapper(PropertiesMapper.class).selectByQuery(query);
   }
 
+  public List<PropertyDto> selectGlobalPropertiesByKeys(DbSession session, Set<String> keys) {
+    return selectByKeys(session, keys, null);
+  }
+
+  public List<PropertyDto> selectPropertiesByKeysAndComponentId(DbSession session, Set<String> keys, long componentId) {
+    return selectByKeys(session, keys, componentId);
+  }
+
+  public List<PropertyDto> selectPropertiesByKeysAndComponentIds(DbSession session, Set<String> keys, Set<Long> componentIds) {
```

Comment:
I have the feeling that at the Mapper level, `selectByKeysAndComponentsIds` and `selectByKeys` could be merged